apps/core/serializers.py

```diff
@@ -165,3 +165,43 @@ def validate(self, attrs):
             )
 
         return attrs
+
+
+class PasswordChangeSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(
+        write_only=True,
+        required=True,
+        validators=[validate_password],
+    )
+    password2 = serializers.CharField(
+        write_only=True,
+        required=True,
+        validators=[validate_password],
+    )
+    old_password = serializers.CharField(
+        write_only=True,
+        required=True,
+        validators=[validate_password],
+    )
+
+    class Meta:
+        model = User
+        fields = ("password", "password2", "old_password", )
+
+    def validate(self, attrs):
+        user = self.context.get("user")
+
+        if attrs["password"] != attrs["password2"]:
+            raise serializers.ValidationError(
+                {"password": "Password fields didn't match"}
+            )
+        if user.check_password(attrs["password"]):
```

Comment:
What happens if the user is None?

Reply:
There will be an `AttributeError`. Added this:
```python
        if not user:
            raise serializers.ValidationError(
                {"user": "The user is not authenticated"}
            )
```

Reply:
this check shoul be here 
https://github.com/jakkemomo/meetups/pull/41/files#diff-7169d74d38e8003c7eb15b3a3a09477e2d09f9131da55babed8dac15d9cf20cbR370